DataBiosphere/leonardo
# IA-3888 Optimize listRuntimeV2 performance

https://broadworkbench.atlassian.net/browse/IA-3888

* Support `creator=role` query string for listRuntimeV2 api
* Update DB query to filter based on an optional `creatorOnly` field for better DB query performance
* Update underlying query to [plain SQL](https://scala-slick.org/doc/3.0.0/sql.html#splicing-literal-values) for better performance (more than 1 min to a few seconds).
* Removed workspace permission check fallback. After talking to @rtitle , it turns out Sam model created by WSM is hierarchical, so by asking sam if a user can read a runtime is sufficient (Sam will take care of checking if the caller has permission at workspace level as well)

Test
1. Run Leo locally against dev DB
2. Run the following cmd as `hermione`
```
curl 'https://local.dsde-dev.broadinstitute.org:30443/api/v2/runtimes?saturnAutoCreated=true&includeLabels=saturnWorkspaceNamespace%2CsaturnWorkspaceName' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H "Authorization: Bearer `gcloud auth print-access-token`" \
  -H 'Cache-Control: no-cache' \
  -H 'Connection: keep-alive' \
  -H 'Origin: http://localhost:3000' \
  -H 'Pragma: no-cache' \
  -H 'Referer: http://localhost:3000/' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: cross-site' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36' \
  -H 'X-App-ID: Saturn' \
  -H 'sec-ch-ua: "Not?A_Brand";v="8", "Chromium";v="108", "Google Chrome";v="108"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  --compressed
```
3. Before the optimization, the DB query will take more than 1 min to finish, and it'll cause curl to timeout. After the optimization, the DB query only takes about 2 seconds to finish. (I put log message before and after the DB query for debugging)

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:


In all cases:
